### ISSUE TITLE
Docs: Add struct tag reference to the README

### ISSUE THREAD

**Issue by Kangaroux** [mixcode/binarystruct#5]
The docs for these are a little buried, having a quick reference would be very helpful. An enumeration of the different struct tags with an example of how to reference other fields for array types is all that's needed. That info already exists in the docs, it's just scattered.

As an aside, syntax highlighting for code blocks in the docs would be a nice bonus.

P.S. I just started using this and I'm liking it so far. Variable length arrays and strings (c-style!) is great. Not being able to always use structs to define payloads with `encoding/binary` really hurts readability, and `protobuf/proto` is way overkill for what I needed.

**Comment by mixcode** [comment]
Yes in recent massive upgrade, we have complete document of tags now. Thank you.
